File: frontend/src/types/session.ts
import { PaymentMethod } from './invoice';

export type PricingModel = 'PER_MINUTE' | 'FLAT_HOURLY' | 'TIME_BLOCK';

export type SessionStatus = 'ACTIVE' | 'PAUSED' | 'COMPLETED' | 'ABANDONED';

export interface SessionResponse {
  id: string;
  seat_id: string;
  member_id: string | null;
  shift_id: string | null;
  status: SessionStatus;
  started_at: string; // ISO datetime
  ended_at: string | null; // ISO datetime
  paused_at: string | null; // ISO datetime
  total_paused_seconds: number;
  locked_rate_paise: number;
  locked_pricing_model: PricingModel;
  package_entitlement_id: string | null;
  promotion_id: string | null;
  discount_paise: number;
  payment_method: PaymentMethod | null;
  created_at: string;
  updated_at: string;
}

File: backend/schemas/session.py
"""Pydantic schemas for GamingSession model."""

from __future__ import annotations

from pydantic import AwareDatetime, Field

from backend.models._enums import PaymentMethod, PricingModel, SessionStatus
from backend.schemas.base import BaseCreateSchema, BaseResponseSchema


class SessionBase(BaseCreateSchema):
    seat_id: str
    member_id: str | None = None
    shift_id: str | None = None
    locked_rate_paise: int = Field(..., ge=0)
    locked_pricing_model: PricingModel
    package_entitlement_id: str | None = None
    promotion_id: str | None = None
    discount_paise: int = Field(0, ge=0)
    payment_method: PaymentMethod | None = None


class SessionCreate(SessionBase):
    pass


class SessionUpdate(BaseCreateSchema):
    member_id: str | None = None
    shift_id: str | None = None
    status: SessionStatus | None = None
    discount_paise: int | None = Field(None, ge=0)
    payment_method: PaymentMethod | None = None


class SessionResponse(SessionBase, BaseResponseSchema):
    id: str
    status: SessionStatus
    started_at: AwareDatetime
    ended_at: AwareDatetime | None = None
    paused_at: AwareDatetime | None = None
    total_paused_seconds: int = 0
    assigned_end_at: AwareDatetime | None = None
    created_at: AwareDatetime
    updated_at: AwareDatetime
